### DISCUSSION TITLE
Why theia did not allow us to provide the best developer experience we could？

### DISCUSSION THREAD

**Discussion by harbin1053020115** [gitpod-io/openvscode-server#341] [Q&A] [answered]
As mentioned in https://www.gitpod.io/blog/cloud-ide-history, it says "Theia did not allow us to provide the best developer experience we could", my question is why theia did not allow us to provide the best developer experience? Is there any limits in theia? Or any other reasons?

**Comment by akosyakov** [comment, marked as answer]
We would like to deliver VS Code in browser (because of its adoption), but since it was not possible, we had to invest in Theia and put a lot of effort to catch up to be "like VS Code". As time passed it turned out to be not necessary anymore so we switched to VS Code.

If you don't need VS Code like experience, but a framework letting you to build own IDE Theia is still a good choice. Just don't expect that everything will work and be aligned with VS Code.

**Reply by harbin1053020115** [reply]
Thanks for your answer. Is there anything unacceptable problems aligned with VS Code in Theia?

**Reply by trusktr** [reply]
@harbin1053020115 a late reply, but maintaining Theia means to keep it up to date with VS Code API changes and additions is a HUGE task. That's basically the explanation. When Theia was made, VS Code didn't run in a browser (or did very poorly with undocumented hacks). Today, VS Code Web runs in web, and is open source, so there isn't much of a benefit of paying many thousands or millions of dollars just to clone something that already works, unless you will make something new and completely different and incompatible from the fork. It is hard to justify that cost.
